### PR TITLE
helm: add pass-credentials key

### PR DESCRIPTION
**Pull request by nkromano**
##### SUMMARY

In helm version [v3.6.1](https://github.com/helm/helm/releases/tag/v3.6.1) when downloading charts from password protected repositories that served from a different domain than the repository, need to use --pass-credentials key.

Add possibility to use the pass-credentials key in helm_repository.py

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

helm_repository

### CODE REVIEW

File: plugins/modules/helm_repository.py [hunk 1]
```diff
@@ -65,6 +65,12 @@
     default: present
     aliases: [ state ]
     type: str
+  pass_credentials:
+    description:
+      - Pass credentials to all domains.
+    required: false
+    default: false
+    type: boolean
```
Comment (abikouo) [inline plugins/modules/helm_repository.py:73]:
```suggestion
    type: boolean
    version_added: 2.3.0
```

File: plugins/modules/helm_repository.py [hunk 2]
```diff
@@ -188,6 +197,7 @@ def main():
             repo_state=dict(
                 default="present", choices=["present", "absent"], aliases=["state"]
             ),
+            pass_credentials=dict(type="bool", default=False)
```
Comment (abikouo) [inline plugins/modules/helm_repository.py:200]:
```suggestion
            pass_credentials=dict(type="bool", default=False),
```

**Review by abikouo — CHANGES_REQUESTED** [review]
@nkromano Thanks for taking the to submit this PR.
You need to attach a changelog.

**Comment by nkromano** [comment]
Hello, @abikouo.
Thanks for comments.
Done.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -4,6 +4,13 @@ Kubernetes Collection Release Notes
 
 .. contents:: Topics
 
+v2.3.0
+======
+
+Minor Changes
+-------------
+
```
Comment (Akasurde) [inline CHANGELOG.rst:12]: Changelog.rst is autogenerated. 
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

File: plugins/modules/helm_repository.py [hunk 4]
```diff
@@ -65,6 +65,13 @@
     default: present
     aliases: [ state ]
     type: str
+  pass_credentials:
+    description:
+      - Pass credentials to all domains.
+    required: false
+    default: false
+    type: boolean
```
Comment (Akasurde) [inline plugins/modules/helm_repository.py:73]:
```suggestion
    type: bool
```

**Review by Akasurde — DISMISSED** [review]

File: plugins/modules/helm_repository.py [hunk 5]
```diff
@@ -157,14 +164,17 @@ def get_repository_status(module, command, repository_name):
 
 # Install repository
 def install_repository(
-    command, repository_name, repository_url, repository_username, repository_password
+    command, repository_name, repository_url, repository_username, repository_password, pass_credentials
```
Comment (gravesm) [inline plugins/modules/helm_repository.py:167]:
```suggestion
    command,
    repository_name,
    repository_url,
    repository_username,
    repository_password,
    pass_credentials,
```

File: plugins/modules/helm_repository.py [hunk 6]
```diff
@@ -219,7 +231,7 @@ def main():
     elif repo_state == "present":
         if repository_status is None:
             helm_cmd = install_repository(
-                helm_cmd, repo_name, repo_url, repo_username, repo_password
+                helm_cmd, repo_name, repo_url, repo_username, repo_password, pass_credentials
```
Comment (gravesm) [inline plugins/modules/helm_repository.py:234]:
```suggestion
                helm_cmd,
                repo_name,
                repo_url,
                repo_username,
                repo_password,
                pass_credentials,
```

**Review by gravesm — COMMENTED** [review]
Looks like the black formatting check is failing for two lines. Otherwise, looks good. Thanks!

**Comment by nkromano** [comment]
Hello, @gravesm!
Done.

File: plugins/modules/helm_repository.py [hunk 7]
```diff
@@ -157,14 +164,22 @@ def get_repository_status(module, command, repository_name):
 
 # Install repository
 def install_repository(
-    command, repository_name, repository_url, repository_username, repository_password
+    command,
+    repository_name,
+    repository_url,
+    repository_username,
+    repository_password,
+    pass_credentials
```
Comment (gravesm) [inline plugins/modules/helm_repository.py:172]:
```suggestion
    pass_credentials,
```

File: plugins/modules/helm_repository.py [hunk 8]
```diff
@@ -219,7 +236,12 @@ def main():
     elif repo_state == "present":
         if repository_status is None:
             helm_cmd = install_repository(
-                helm_cmd, repo_name, repo_url, repo_username, repo_password
+                helm_cmd,
+                repo_name,
+                repo_url,
+                repo_username,
+                repo_password,
+                pass_credentials
```
Comment (gravesm) [inline plugins/modules/helm_repository.py:244]:
```suggestion
                pass_credentials,
```

**Review by gravesm — COMMENTED** [review]
There are still two trailing commas missing which is causing the black check to fail. Once these are fixed and CI passes we can get this merged.

**Comment by gravesm** [comment]
recheck

**Comment by Akasurde** [comment]
recheck

**Comment by gravesm** [comment]
recheck

**Review by gravesm — APPROVED** [review]

**Pull request merged**